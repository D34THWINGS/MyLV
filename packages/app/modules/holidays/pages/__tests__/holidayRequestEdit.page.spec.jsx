import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { HolidayRequestEditPage } from '../holidayRequestEdit.page'

jest.unmock('../holidayRequestEdit.page')

describe('HolidayRequestEditPage', () => {
  let props
  beforeEach(() => {
    props = {
      putHoliday: jest.fn(),
      holidayRequest: {
        user: 'foo',
        periods: [{
          startDate: '2012-01-02',
          endDate: '2012-01-03',
        }],
      },
    }
  })

  const getWrapper = () => shallow(<HolidayRequestEditPage {...props} />)

  it('should render properly', () => {
    // When
    const wrapper = shallow(getWrapper().prop('render')({ children: <div>Foo</div> }))

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
