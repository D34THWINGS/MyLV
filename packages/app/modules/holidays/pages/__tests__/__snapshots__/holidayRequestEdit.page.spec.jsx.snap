// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HolidayRequestEditPage should render properly 1`] = `
<Card
  classes={
    Object {
      "root": "MuiCard-root-1",
    }
  }
  raised={false}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Editer une demande de congé | 
      MyLV
    </title>
  </HelmetWrapper>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      Editer une demande de congé
    </WithStyles(Typography)>
    <div>
      Foo
    </div>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)>
    <WithStyles(Button)
      color="primary"
      disabled={true}
      size="small"
      type="submit"
    >
      Sauvegarder
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</Card>
`;
